spec/lib/sendgrid_actionmailer_spec.rb

```diff
@@ -341,100 +337,77 @@ def post(request_body:)
           expect(client.sent_mail['ip_pool_name']).to eq('marketing')
         end
 
-        context 'parse object' do
-          it "should parse 1.8 hash" do
-            asm = {'group_id' => 99, 'groups_to_display' => [4,5,6,7,8]}
-            mail['asm'] = asm
-            mailer.deliver!(mail)
-            expect(client.sent_mail['asm']).to eq({"group_id" => 99, "groups_to_display" => [4,5,6,7,8]})
-          end
-
-          it "should parse 1.9 hash" do
-            asm = { group_id: 99, groups_to_display: [4,5,6,7,8]}
-            mail['asm'] = asm
-            mailer.deliver!(mail)
-            expect(client.sent_mail['asm']).to eq({"group_id" => 99, "groups_to_display" => [4,5,6,7,8]})
-          end
-
-          it "should parse json" do
-            asm = {'group_id' => 99, 'groups_to_display' => [4,5,6,7,8]}
-            mail['asm'] = asm.to_json
-            mailer.deliver!(mail)
-            expect(client.sent_mail['asm']).to eq({"group_id" => 99, "groups_to_display" => [4,5,6,7,8]})
-          end
-
-          it 'should not change values inside custom args' do
-            custom_args = { 'text' => 'line with a => in it' }
-            mail['custom_args'] = custom_args
-            mailer.deliver!(mail)
-            expect(client.sent_mail['custom_args']).to eq('text' => 'line with a => in it')
-          end
```

Comment:
Sweet, red diff-lines :D

Reply:
Yeah, these use cases - 1.8 style hash and json hash - were not outlined in the Readme. Only supporting the 1.9 hash usage outlined in the Readme simplified the the code immensely. So axed.